src/ripple/app/sidechain/impl/SignatureCollector.cpp

```diff
@@ -139,9 +139,41 @@ void
 SignatureCollector::expire()
 {
     std::lock_guard lock(mtx_);
+    // Never expire collections with this servers sig or submitted txns
+    for (auto i = messages_.begin(), e = messages_.end(); i != e; ++i)
+    {
+        auto const& multiSigMsg = i->second;
+        if (multiSigMsg.submitted_)
+        {
+            messages_.touch(i);
+            continue;
+        }
+        for (auto const& [pk, sig] : multiSigMsg.sigMaps_)
+        {
+            if (pk == myPubKey_)
+            {
+                messages_.touch(i);
```

Comment:
add `break;` after the `touch(i)`?

Reply:
or replace the whole second loop with something like:
```
if (std::any_of(multiSigMsg.sigMaps_.begin(), multiSigMsg.sigMaps_.end(),
                [&](const auto& p) { return p.first == myPubKey_; })) 
{
    messages_.touch(i);
}
```

Reply:
Breaking after the touch is a good idea. Will do. I prefer a raw loop over `any_of` here, but that's just a personal preference; both are equally easy to read.

Reply:
Yeah, the `continue` in one and a `break` in another is a good argument. I'll make the change.